crates/transaction-pool/src/pool/mod.rs

```diff
@@ -242,6 +245,15 @@ where
         self.transaction_listener.lock().push(listener);
         rx
     }
+    /// Adds a new blob sidecar listener to the pool that gets notified about every new
+    /// eip4844 transaction's blob sidecar.
+    pub fn add_blob_sidecar_listener(&self) -> mpsc::Receiver<BlobTransactionSidecar> {
```

Comment:
this is an issue because the sidecar doesn't have any identifying information.

I think we want a new type here that includes the sidecar and tx hash

Reply:
makes sense, thanks for clarifying, will change this.


> assumptions here include
stream consumer doesn't need the correspinding TxHash for a given blob sidecar

so the consumer will need the corresponding txhash to identify which tx the sidecar came from

Reply:
taken care of with the `NewBlobSidecar` type that's defined